### PR TITLE
8217 move extract_qb_data to its own class

### PR DESCRIPTION
**Pull request by arisah-ijeoma**
https://github.com/sassafrastech/madeline/pull/478 needs to be merged in first

### CODE REVIEW

File: spec/models/accounting/quickbooks/data_extractor_spec.rb [hunk 1]
```diff
@@ -0,0 +1,162 @@
+require 'rails_helper'
+
+describe Accounting::Quickbooks::DataExtractor, type: :model do
+  let(:generic_service) { instance_double(Quickbooks::Service::ChangeDataCapture, since: double(all_types: [])) }
```
Comment (meltheadorable) [inline spec/models/accounting/quickbooks/data_extractor_spec.rb]: This shouldn't be necessary

File: spec/models/accounting/quickbooks/data_extractor_spec.rb [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  let(:qb_id) { 1982547353 }
+  let(:division) { create(:division, :with_accounts) }
+  let(:prin_acct) { division.principal_account}
+  let(:int_inc_acct) { division.interest_income_account }
+  let(:int_rcv_acct) { division.interest_receivable_account }
+  let(:txn_acct) { create(:account, name: 'Some Bank Account') }
+  let(:loan) { create(:loan, division: division) }
+  let(:journal_entry) { instance_double(Quickbooks::Model::JournalEntry, id: qb_id, as_json: quickbooks_data) }
```
Comment (meltheadorable) [inline spec/models/accounting/quickbooks/data_extractor_spec.rb]: I don't think you need this either.

**Comment by arisah-ijeoma** [comment]
the pr that follows this #501 has updates to the tests @meltheadorable

**Comment by emitche** [comment]
@arisah-ijeoma: As this was marked as `needs-minor-revision` is it okay if this just gets moved on to QA instead of being reviewed again? Or do you prefer this be reviewed? Thanks!

**Comment by arisah-ijeoma** [comment]
@emitche, it can move to qa

**Comment by emitche** [comment]
@arisah-ijeoma Can you resolved these merge conflicts? I want to make sure the right updates are applied. Thank you!

**Pull request merged**